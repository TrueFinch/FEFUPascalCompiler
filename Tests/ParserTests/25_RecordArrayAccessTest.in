program Project1;

type 
    a = record
        k: array [0..5] of integer;
        l: integer;
    end;
  
var 
    i: a;

begin
    i.k[0] := 0;
    i.l := 4;
    write(i.k[2]);
    write(i.l);
end.