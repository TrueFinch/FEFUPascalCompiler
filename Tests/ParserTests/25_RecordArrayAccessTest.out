                                                                project1                                                                 
                                                                   ┼                                                                     
                                                               MainBlock                                                                 
                   ┌──────────────────────────┬────────────────────┴──────────────────────────┐                                          
                 type                        var                                      CompoundStatement                                  
                   ┼                          ┼                ┌──────────────────┬───────────┴───────┬──────────────────────┐           
               TypeDecl                 SimpleVarDecl          :=                :=              FunctionCall          FunctionCall      
 ┌─────────────────┴┐                   ┌─────┴┐              ┌┴────────┐       ┌─┴──────┐    ┌───────┴───┐           ┌──────┴──┐        
 a              RecordType              i  SimpleType    ArrayAccess    0  RecordAccess  4  write    ArrayAccess    write  RecordAccess  
             ┌──────┴──────────┐           ┌───┴            ┌─┴──────┐     ┌──┬─┴                       ┌─┴──────┐         ┌──┬─┴        
             :                 :           a           RecordAccess  0     i  l                    RecordAccess  2         i  l          
    ┌────────┴┐          ┌─────┴┐                      ┌──┬─┴                                      ┌──┬─┴                                
    k     ArrayType      l  SimpleType                 i  k                                        i  k                                  
        ┌─────┴──┐             ┌┴                                                                                                        
        ..   SimpleType     integer                                                                                                      
       ┌┴─┐     ┌┴                                                                                                                       
       0  5  integer                                                                                                                     